### PR TITLE
Document public functions

### PR DESCRIPTION
**Pull request by folkertdev**

### CODE REVIEW

**Comment by bjorn3** [comment]
This PR has a bunch of conflicts that need to be resolved.

File: bzlib.rs [hunk 1]
```diff
@@ -464,9 +453,21 @@ impl<T> DSlice<T> {
...
+///
+/// And destructed by:
+///
+/// - [`BZ2_bzReadClose`]
+/// - [`BZ2_bzWriteClose`]
+/// - [`BZ2_bzclose`]
 #[allow(non_camel_case_types)]
-#[derive(Copy, Clone)]
-pub(crate) struct bzFile {
+pub struct BZFILE {
```
[the remark this answers is not in the document]
Comment (bjorn3) [reply bzlib.rs:470]: `bzFile` is not a public type. Also I think you slightly messed up the rebase. You removed the `pub(crate)` I had added to it again.

[the remark this answers is not in the document]
Comment (folkertdev) [reply bzlib.rs:470]: no that's correct, `BZFILE` is now public (it's a `#define BZFILE void` in the C source)

[the remark this answers is not in the document]
Comment (folkertdev) [reply bzlib.rs:470]: so, then, leave it like this? from C it doesn't make a difference, from rust this actually makes for a more accurate signature.

**Review by bjorn3 — APPROVED** [review]

**Pull request merged**